simply/actor.py

```diff
@@ -177,27 +157,32 @@ def __init__(self, actor_id, df, battery=None, csv=None, ls=1, ps=1, pm={}, clus
         self.args = {"id": actor_id, "df": df.to_json(), "csv": csv, "ls": ls, "ps": ps,
                      "pm": pm}
 
+    def get_mm_buy_prices(self):
+        env = self.environment
+        grid_fee = env.get_grid_fee(bid_cluster=env.market_maker.cluster, ask_cluster=self.cluster)
+        # the achievable prices the mm buys energy for from the actor are reduced by the grid fee
+        return env.market_maker.buy_prices-grid_fee
+    # creating a property object
+    mm_buy_prices = property(get_mm_buy_prices)
+
+    def get_mm_sell_prices(self):
+        env = self.environment
+        grid_fee = env.get_grid_fee(ask_cluster=env.market_maker.cluster, bid_cluster=self.cluster)
+        # the prices for which the mm sells energy to the actor are increased by the grid fee
+        return self.environment.market_maker.sell_prices+grid_fee
+
+    # creating a property object
+    mm_sell_prices = property(get_mm_sell_prices)
+
+    def get_t_step(self):
+        return self.environment.time_step
```

Comment:
is this used somewhere? also all components that have access to the market also should have access to the environment directly

Reply:
it is used alot. Its set to a property a few lines below which makes accessing the current time
`self.t_step`
instead of
`self.environment.time_step`
Also if we want to change the source of the timestep only this line has to be changed